sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -283,7 +284,7 @@ class HadoopTableReader(
       _minSplitsPerRDD)
 
     // Only take the value (skip the key) because Hive works only with values.
-    rdd.map(_._2)
+    CombineSmallFile.combineWithPath(rdd, sc, path).map(_._2)
```

Comment:
What about the default value of `spark.sql.small.file.combine` is false, and when user find there're many small files in table, `set spark.sql.small.file.partition.number=x` instead of calculate the partitionNum.